### PR TITLE
Sup3 axios json server

### PR DESCRIPTION
**Pull request by kuryloff**

### CODE REVIEW

File: sign-up-task/src/Utils/validators.js [hunk 1]
```diff
@@ -37,5 +39,32 @@ export const fieldValidator = (fieldName, value = '', passwordValue = '') => {
...
+export const signUpFormValidator = async (newUser, email) => {
+    let users = await signUpAPI.getUsers();
+    let isUser = users.some(user => user.email === email)
+    isUser && signUpAPI.setUser(newUser)
+    return isUser;
 }
+
+export const signInFormValidator = async (input, email, password) => {
+    let users = await signUpAPI.getUsers();
+    let user = emailValidator(users, email) || ""
```
Comment (svyatoslav00012) [inline sign-up-task/src/Utils/validators.js:53]: why you need user here?

File: sign-up-task/src/Utils/validators.js [hunk 2]
```diff
@@ -37,5 +39,32 @@ export const fieldValidator = (fieldName, value = '', passwordValue = '') => {
...
+}
 
+export const signUpFormValidator = async (newUser, email) => {
+    let users = await signUpAPI.getUsers();
+    let isUser = users.some(user => user.email === email)
+    isUser && signUpAPI.setUser(newUser)
+    return isUser;
 }
+
+export const signInFormValidator = async (input, email, password) => {
```
Comment (svyatoslav00012) [inline sign-up-task/src/Utils/validators.js:51]: why you need input here?

File: sign-up-task/src/Utils/validators.js [hunk 3]
```diff
@@ -37,5 +39,32 @@ export const fieldValidator = (fieldName, value = '', passwordValue = '') => {
...
+}
+
+const emailValidator = (users, email) => {
+    let userToCheck;
+    users.forEach(user => user.email === email && (userToCheck = user))
+    return userToCheck;
+};
+
+
+const passwordValidator = (user, password) => (user.password === password)
```
Comment (svyatoslav00012) [inline sign-up-task/src/Utils/validators.js:70]: you better to bcrypt password before saving, and then check using bcrypt

File: sign-up-task/src/Utils/validators.js (hunk 1 above)
Comment (IhorKurylov) [reply sign-up-task/src/Utils/validators.js:53]: спростити через :
let user = emailValidator(await signUpAPI.getUsers(), email) || ""

File: sign-up-task/src/Utils/validators.js (hunk 2 above)
Comment (kuryloff) [reply sign-up-task/src/Utils/validators.js:51]: garbage is left after refactoring((
mu fault

File: sign-up-task/src/Utils/validators.js (hunk 3 above)
Comment (kuryloff) [reply sign-up-task/src/Utils/validators.js:70]: will try to bcrypt. I have no experience with it

File: sign-up-task/src/Utils/validators.js (hunk 1 above)
Comment (kuryloff) [reply sign-up-task/src/Utils/validators.js:53]: will try to refactor, thanks))

**Pull request merged**